packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -1016,12 +1016,56 @@ describeSpec('Listens:', [], () => {
           // are unlistened by now.
           .userUnlistens(limitToLast)
           .client(0)
-          .expectListen(limitToLast)
+          // TODO(b/143693491) `expectListen` would also pass, which is wrong.
+          // the subsequent `expectActiveTargets` makes sure the target is
+          // removed, but we still need to fix `spec_test_runner`.
+          .expectUnlisten(limitToLast)
           .expectActiveTargets()
       );
     }
   );
 
+  specTest('Mirror Queries in primary client', [], () => {
+    const limit = Query.atPath(path('collection'))
+      .addOrderBy(orderBy('val', 'asc'))
+      .withLimitToFirst(2);
+    const limitToLast = Query.atPath(path('collection'))
+      .addOrderBy(orderBy('val', 'desc'))
+      .withLimitToLast(2);
+    const docA = doc('collection/a', 1000, { val: 0 });
+    const docB = doc('collection/b', 1000, { val: 1 });
+    const docC = doc('collection/c', 2000, { val: 0 });
+
+    return (
+      spec()
+        .userListens(limit)
+        .expectListen(limit)
+        .userListens(limitToLast)
+        .expectListen(limitToLast)
+        .watchAcksFull(limit, 1000, docA, docB)
+        .expectEvents(limit, { added: [docA, docB] })
+        .expectEvents(limitToLast, { added: [docB, docA] })
+        // Un-listen to limitToLast
+        .userUnlistens(limitToLast)
+        .expectUnlisten(limitToLast)
+        .watchSends({ affects: [limit] }, docC)
+        .watchSnapshots(2000)
+        .expectEvents(limit, { added: [docC], removed: [docB] })
+        // Re-listens to limitToLast
+        .userListens(limitToLast)
+        .expectListen(limitToLast)
```

Comment:
This is kind of strange since it should either not exist or have a resume token. Can you add a comment to b/143693491 that we should clean this up (ideally by removing the expectListen).

Reply:
We can remove `expectListen` here, because it is actually checking actual target, as explained above. 

I want to keep it still, such that when `expectListen` gets fixed, this can be one of the case to cover.